src/arrayMethodSort.js

```diff
@@ -4,9 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(callback = compareAsStrings) {
+    let count = 0;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const prev = this[i - 1];
+        const current = this[i];
+
+        if (callback(prev, current) > 0) {
+          count++;
+          this[i - 1] = current;
+          this[i] = prev;
+        }
+      }
+    } while (count > 0);
+
+    return this;
   };
 }
 
+const compareAsStrings = (a, b) => {
+  const stringA = String(a);
+  const stringB = String(b);
+
+  if (stringA > stringB) {
+    return 1;
+  } else if (stringA === stringB) {
+    return 0;
+  } else {
+    return -1;
```

Comment:
try to avoid `if else `construction

Reply:
In this case, the use of such a construction is the best option, since replacement by a ternary operator is not possible here, and the use of several "if" operators entails unnecessary checks.

Reply:
fixed